stdlib/public/core/HashedCollections.swift.gyb

```diff
@@ -2473,6 +2473,11 @@ collections = [
 /// keys, followed by the values.
 //
 // See the docs at the top of the file for more details on this type
+//
+// NOTE: The precise layout of this type is relied on in the runtime
+// to provide a statically allocated empty singleton. 
```

Comment:
Right, I was asking about this on swift-dev. Is fixed-layout sufficient to prevent reordering, or do I need something stronger  to match the C abi?